lib/bolt/plugin/puppetdb.rb

```diff
@@ -45,8 +46,19 @@ def resolve_reference(opts)
         targets = @puppetdb_client.query_certnames(opts['query'])
         facts = []
 
-        target_opts = opts.select { |k, _| TARGET_OPTS.include?(k) }
-        Bolt::Util.walk_vals(target_opts) do |value|
+        template = opts.delete('target_mapping') || {}
+
+        keys = Set.new(TEMPLATE_OPTS) & opts.keys
+        unless keys.empty?
+          raise Bolt::ValidationError, "PuppetDB plugin expects keys #{keys} to be set under 'target_mapping'"
```

Comment:
@beechtom I don't particularly mind this but I suspect you might: printing sets in Ruby isn't the greatest:
```
lucy ~/githubs/bolt ((e07533e...)) $ be bolt command run 'hostname' -t all
Error executing plugin puppetdb from resolve_reference in puppetdb: PuppetDB plugin expects keys #<Set: {"name", "config"}> to be set under 'target_mapping'
```

You can `to_a` for slightly better printing:
```
lucy ~/githubs/bolt ((e07533e...)) $ be bolt command run 'hostname' -t all
Error executing plugin puppetdb from resolve_reference in puppetdb: PuppetDB plugin expects keys ["name", "config"] to be set under 'target_mapping'
```